src/main/java/us/kbase/userprofile/MongoController.java

```diff
@@ -154,39 +148,60 @@ public List<User> filterUsers(String filter) {
 	}
 	
 	private void ensureIndex() {
-		DBObject userUnique = new BasicDBObject("user.username",1);
+		Document userUnique = new Document("user.username",1);
 		//DBObject userText = new BasicDBObject("user.username","text");
-		DBObject unique = new BasicDBObject("unique",1);
+		IndexOptions uniqueOptions = new IndexOptions().unique(true);
 		//DBObject nameText = new BasicDBObject("user.realname","text");
-		profiles.createIndex(userUnique, unique);
+		profiles.createIndex(userUnique, uniqueOptions);
 		//profiles.ensureIndex(userText);
 		//profiles.ensureIndex(nameText);
 	}
-	
-	
+
 	public boolean exists(String username) {
-		DBObject query = new BasicDBObject("user.username",username);
-		if(profiles.findOne(query, new BasicDBObject())==null) {
-			return false;
+		return findProfileByUsername(username) != null;
+    }
+
+	private Document findProfileByUsername(String username) {
+		Document query = new Document("user.username", username);
+		return profiles.find(query).first();
+	}
+
+	private Document createMergedProfile(String username, JsonNode profileNode) {
+		Document mergedProfile = new Document();
+		Document oldDoc = findProfileByUsername(username);
+		if (oldDoc.containsKey("profile") && oldDoc.get("profile") != null) {
+			mergedProfile = new Document(oldDoc.get("profile", Document.class));
+		}
+
+		Document newDoc = Document.parse(profileNode.toString());
```

Comment:
Probably cheaper to transform it to a Map vs. serialize & deserialize via a string

Reply:
I agree, but this entire file, along with the test file, serializes and deserializes using a string. It will require another PR if we decide to make the conversion.

Reply:
For now, if the code has an antipattern and you don't touch that part of the code, don't worry about it
If you make minor modifications (say whitespace etc) I leave it up to you
But if you're creating a whole new method don't add new instances of antipatterns

Reply:
obsolete

Reply:
:+1: